pep-0678.rst

```diff
@@ -229,27 +245,86 @@ proposed ``__note__`` semantics, but this would be rarely and inconsistently
 applicable.
 
 
-Store notes in ``ExceptionGroup`` s
------------------------------------
+Store notes in ``ExceptionGroup``\ s
+------------------------------------
 Initial discussions proposed making a more focussed change by thinking about how to
 associate messages with the nested exceptions in ``ExceptionGroup`` s, such as a list
 of notes or mapping of exceptions to notes.  However, this would force a remarkably
 awkward API and retains a lesser form of the cross-referencing problem discussed
-under "use ``print()`` " above; if this PEP is rejected we prefer the status quo.
+under "use ``print()``" above; if this PEP is rejected we prefer the status quo.
 Finally, of course, ``__note__`` is not only useful with ``ExceptionGroup`` s!
 
 
 
+Possible Future Enhancements
+============================
+
+In addition to rejected alternatives, there have been a range of suggestions which
+we believe should be deferred to a future version, when we have more experience with
+the uses (and perhaps misuses) of ``__note__``.
+
+
+Allow any object, and cast to string for display
+------------------------------------------------
+We have not identified any scenario where libraries would want to do anything but either
+concatenate or replace notes, and so the additional complexity and interoperability
+challenges do not seem justified.
+
+Permitting any object would also force any future structured API to change the behaviour
+of already-legal code, whereas expanding the legal contents of ``__note__`` from strings
+to include other objects is fully backwards-compatible.  In the absence of any proposed
+use-case (see also `Non-goals`_), we strongly prefer to keep things simple.
+
+
+Add a helper function ``contexlib.add_exc_note()``
+--------------------------------------------------
+We don't expect notes to be used frequently enough to justify this either, but
+provide the following extensible recipe:
```

Comment:
```suggestion
It was suggested that we add a utility such as the one below to the standard
library. We are open to this idea, but do not see it as a core part of the
proposal of this PEP as it can be added as an enhancement later.
```